runtimes/pythonrt/py.go

```diff
@@ -143,66 +143,88 @@ func pyExeInfo(ctx context.Context, exePath string) (exeInfo, error) {
 }
 
 type pyRunInfo struct {
-	rootDir  string
-	sockPath string
-	lis      net.Listener
-	proc     *os.Process
+	userRootDir string
+	pyRootDir   string
+	sockPath    string
+	lis         net.Listener
+	proc        *os.Process
+}
+
+func (ri *pyRunInfo) Cleanup() {
+	if ri.userRootDir != "" {
+		os.RemoveAll(ri.userRootDir)
+	}
+
+	if ri.pyRootDir != "" {
+		os.RemoveAll(ri.pyRootDir)
+	}
 }
 
 type runOptions struct {
 	log            *zap.Logger
-	pyExe          string
-	tarData        []byte
-	rootPath       string
-	env            map[string]string
+	pyExe          string            // Python executable to use
+	tarData        []byte            // tar data from build stage
+	rootPath       string            // simple.py:greet
+	env            map[string]string // Python process environment
 	stdout, stderr io.Writer
 }
 
 const runnerMod = "ak_runner"
 
 func runPython(opts runOptions) (*pyRunInfo, error) {
-	rootDir, err := os.MkdirTemp("", "ak-")
-	opts.log.Info("root dir", zap.String("path", rootDir))
+	runOK := false
+	var ri pyRunInfo
+	var err error
+	defer func() {
+		if !runOK {
+			ri.Cleanup()
+		}
+	}()
+
+	ri.userRootDir, err = os.MkdirTemp("", "ak-")
```

Comment:
Question: do we actually care / need to identify our temp dirs? I'm not strictly against it, just wondering if it has a useful purpose at this point.

Reply:
When debugging it's easy to find these directories.